test/dynamo/test_exitstack.py

```diff
@@ -0,0 +1,588 @@
+# Owner(s): ["module: dynamo"]
+import contextlib
+import sys
+import traceback
+import unittest
+from contextlib import contextmanager, ExitStack
+
+import torch
+import torch._dynamo.test_case
+
+from .utils import make_dynamo_test
+
+
+@contextlib.contextmanager
+def set_default_dtype(dtype):
+    old_dtype = torch.get_default_dtype()
+    try:
+        torch.set_default_dtype(dtype)
+        yield
+    finally:
+        torch.set_default_dtype(old_dtype)
+
+
+class TestExitStack(torch._dynamo.test_case.TestCase):
+    def setUp(self):
+        if sys.version_info < (3, 11):
+            self.skipTest(
+                "Tracing the unittest module needs exception table (Python 3.11+) to work"
+            )
+        self._old = torch._dynamo.config.enable_trace_contextlib
+        torch._dynamo.config.enable_trace_contextlib = True
+
+    def tearDown(self):
+        torch._dynamo.config.enable_trace_contextlib = self._old
+
+    def test_exitstack(self):
+        @torch.compile(backend="eager", fullgraph=True)
+        def fn(t):
+            with contextlib.ExitStack() as stack:
+                stack.enter_context(set_default_dtype(torch.float64))
+                return t.sin()
+
+        t = torch.randn(2, dtype=torch.float64)
+        y = fn(t)
+        self.assertEqual(y, t.sin())
+        self.assertEqual(y.dtype, torch.float64)
+
+
+class CPythonTestBaseExitStack:
```

Comment:
Missing link to CPython https://github.com/python/cpython/blob/v3.13.1/Lib/test/test_contextlib.py